### PR TITLE
Rename blacklist/whitelist to more accurate and appropriate names.

### PR DESCRIPTION
**Pull request by benlesh**
This PR renames all uses of "blacklist" and "whitelist" to more accurate and appropriate names.

The Angular team is committed to making sure this is an inclusive environment for all. This PR is not about the origins of the phrases, rather it's about making everyone feel welcome and comfortable while reading our code base.

### CODE REVIEW

**Comment by SebastianPodgajny** [comment]
That's amazing, but why you want to merge it to MASTER branch? I find it offensive 😞 and there is a lot of `master` in this repository MasterService, MasterFlag, MasterConfig it is notappropriatee name in 2019 in my opinion. Maybe we should change it to leader (leader branch, leader service etc) that is more welcoming.
Also `RubberMan` from ToH should be changed to `RubberPerson` also `son` it's not ideal part maybe `RubberPerchild` that's better 🙂

**Comment by skreborn** [comment]
Personal opinions aside - as it is up to the Angular team to decide their preferences on this matter -, I don't believe "safelist" is a viable alternative to "whitelist". The latter word is simply more versatile in meaning, while the former makes it difficult to understand its purpose in some cases.

A whitelisted item can give you access, permissions, exception, safety, and a number of other things. On the other hand, a safelist implies only one of those.

**Comment by SebastianPodgajny** [comment]
Of Course that is only my opinion and Angular team will decide, but i wanted to express it as it relates to this topic

**Comment by alfaproject** [comment]
I respect and commend any effort on inclusivity. I'm glad more organizations are doing it, but in this particular case, isn't it a bit too farfetched?

In any case, why not consider other alternatives for some of those cases like `includes`/`excludes` since it's basically what those lists are for: either to include and accept something or exclude and not accept something. `block`/`safe` seem to have a stronger semantic meaning that doesn't really suit in a lot of those replacements. I don't think search/replace is a good thing for this change.

**Comment by ericmartinezr** [comment]
I would label it as "risk: high" since if sjw dont like this all of our careers and our phisycal integrity are in great danger

**Comment by jenniferfell** [comment]
> I respect and commend any effort on inclusivity. I'm glad more organizations are doing it, but in this particular case, isn't it a bit too farfetched?
> 
> In any case, why not consider other alternatives for some of those cases like `includes`/`excludes` since it's basically what those lists are for: either to include and accept something or exclude and not accept something. `block`/`safe` seem to have a stronger semantic meaning that doesn't really suit in a lot of those replacements. I don't think search/replace is a good thing for this change.

To be clear, my thumbs up here is not that this request is farfetched. I think it's a reasonable request. My thumbs up is to include/exclude, which I agree are more neutral in tone than block/safe. But I defer to the technical intent re the strength of terms chosen.

**Comment by robwormald** [comment]
Reminder to all on Angular's [Code of Conduct](https://github.com/angular/code-of-conduct/blob/master/CODE_OF_CONDUCT.md) 

> Communication through any of Angular's channels (GitHub, Gitter, IRC, mailing lists, Google+, Twitter, etc.) must be constructive and never resort to personal attacks, **trolling**, public or private harassment, insults, or other unprofessional conduct.

Please keep all feedback on-topic and constructive.

**Comment by SebastianPodgajny** [comment]
Yes my comment was a joke, but I want to make a point by showing where it is going, blacklist is blacklist word used in industry for years and I don't see anything wrong with it. I really admire all of you and yours work on this framework, please don't take this framework into this political games.

Why `black` in a word is not welcoming and not comfortable?
Is blacksmith, blackmail and blackjack banned words in USA already?

**Comment by robwormald** [comment]
@SebastianPodgajny if you'd like to discuss this further, you are welcome to email me - robwormald at google dot com, and I'm happy to explain our rationale behind making changes like this, and also why comments and trolling like yours are unhelpful and counterproductive to our project and community. 

Pull Requests are not the place. Final Warning.

**Comment by benlesh** [comment]
The readability of the variable names we choose is a valid concern. However we cannot return to "whitelist" or "blacklist", that's not an option.

Going forward, we will just need to use names that are appropriate that are _not_ "whitelist" or "blacklist". So we might have "approvedList", "bannedList", "safeList", "accessList", etc. Whatever makes the most sense.

To reiterate, **"blacklist" and "whitelist" are off the table, and will not return**. I think we can come up with better, more descriptive names for such lists anyhow.

**Comment by alfaproject** [comment]
@benlesh that's basically my point. I don't think a blind search/replace will do any good. If it's being changed now, please take a proper look at the contextual uses of the replacements.

P.S.: `blocklist` is really bad by the way and it almost sounds like you know what.

**Comment by benlesh** [comment]
presubmit at:  http://test/OCL:234037925:BASE:234037977:1550185741537:eb43f3d1

File: packages/platform-browser/test/dom/events/event_manager_spec.ts [hunk 1]
```diff
@@ -296,7 +296,7 @@ import {el} from '../../../testing/src/browser_util';
       expect(receivedEvents).toEqual([]);
     });
 
-    it('should run blockListedEvents handler outside of ngZone', () => {
+    it('should run disallowedEvents handler outside of ngZone', () => {
```
Comment (skreborn) [inline packages/platform-browser/test/dom/events/event_manager_spec.ts]: It's called "disabled" above, but here it's "disabled". Aren't those two the same? If so, we should probably use only one of these terms (preferably "disallowed").

File: packages/platform-browser/testing/src/browser_util.ts [hunk 2]
```diff
@@ -107,7 +107,7 @@ export function normalizeCSS(css: string): string {
       .replace(/\[(.+)=([^"\]]+)\]/g, (...match: string[]) => `[${match[1]}="${match[2]}"]`);
 }
 
-const _singleTagWhitelist = ['br', 'hr', 'input'];
+const _safeTags = ['br', 'hr', 'input'];
```
Comment (skreborn) [inline packages/platform-browser/testing/src/browser_util.ts]: Should call this `_emptyTags` or `_voidTags`, as single-tag elements are usually either called empty or void elements.

File: test-events.js [hunk 3]
```diff
@@ -6,4 +6,4 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
-Zone[Zone.__symbol__('BLACK_LISTED_EVENTS')] = ['scroll'];
```
Comment (skreborn) [inline test-events.js]: zone.js support

File: tools/material-ci/angular_material_test_blocklist.js [hunk 4]
```diff
@@ -16,7 +16,7 @@
 // clang-format off
 // tslint:disable
 
-window.testBlocklist = {
+window.testBlockList = {
```
Comment (skreborn) [inline tools/material-ci/angular_material_test_blocklist.js]: Why was the capitalization changed here?

**Review by skreborn — CHANGES_REQUESTED** [review]
I believe this is going in a much better direction than initially, but there are still a few things to look at.

- There are a _lot_ of unnecessary white-space changes included in this PR. I'm guessing that this is due to an automatic formatter, but maybe we should consider excluding them unless they improve upon the code.

-  Zone.js-private variables are renamed but these changes are not supported on the [other side](https://github.com/angular/zone.js/blob/master/lib/browser/browser.ts#L56).

- There is also something going on with the polyfills file, which I just don't understand at all. Why were those changes made at all? Why the separation of files, and why did you extend the description? I believe it was perfectly good as it was, but I'm open to a conversation if you have a valid argument.

File: aio/tools/examples/shared/boilerplate/cli/src/polyfills.ts [hunk 5]
```diff
@@ -60,26 +60,38 @@ import 'core-js/es7/reflect';
...
  * user can disable parts of macroTask/DomEvents patch by setting following flags
- */
-
- // (window as any).__Zone_disable_requestAnimationFrame = true; // disable patch requestAnimationFrame
- // (window as any).__Zone_disable_on_property = true; // disable patch onProperty such as onclick
- // (window as any).__zone_symbol__BLACK_LISTED_EVENTS = ['scroll', 'mousemove']; // disable patch specified eventNames
-
- /*
- * in IE/Edge developer tools, the addEventListener will also be wrapped by zone.js
+ * because those flags need to be set before `zone.js` being loaded, and webpack
```
Comment (IgorMinar) [inline aio/tools/examples/shared/boilerplate/cli/src/polyfills.ts]: has this change been upstreamed to the CLI repo?

Comment (IgorMinar) [reply aio/tools/examples/shared/boilerplate/cli/src/polyfills.ts]: I think @skreborn is right that this change seems unnecessary. can you explain?

File: aio/tools/transforms/angular-base-package/processors/createSitemap.js [hunk 6]
```diff
@@ -1,15 +1,11 @@
...
       'navigation-json',
       'contributors-json',
       'resources-json',
     ],
-    blacklistedPaths: [
-      'test',
-      'file-not-found',
-      'overview-dump'
-    ],
+    ignoredPaths: ['test', 'file-not-found', 'overview-dump'],
```
[the remark this answers is not in the document]
Comment (IgorMinar) [reply aio/tools/transforms/angular-base-package/processors/createSitemap.js]: wild guess: line length limit caused the first one to be formatted differently?

File: aio/tools/transforms/angular-base-package/processors/createSitemap.js [hunk 7]
```diff
@@ -19,15 +15,15 @@ module.exports = function createSitemap() {
...
-          // Filter out docs that are not outputted
-          .filter(doc => doc.outputPath)
-          // Filter out unwanted docs
-          .filter(doc => this.blacklistedDocTypes.indexOf(doc.docType) === -1)
-          .filter(doc => this.blacklistedPaths.indexOf(doc.path) === -1)
-          // Capture the path of each doc
-          .map(doc => doc.path)
-          // Convert the homepage: `index` to `/`
-          .map(path => path === 'index' ? '' : path)
+                  // Filter out docs that are not outputted
```
[the remark this answers is not in the document]
Comment (IgorMinar) [reply aio/tools/transforms/angular-base-package/processors/createSitemap.js]: yeah.. it's unclear what or how this file was reformatted.

File: aio/tools/transforms/angular-base-package/processors/createSitemap.spec.js [hunk 8]
```diff
@@ -39,45 +35,45 @@ describe('createSitemap processor', () => {
...
+          {path: 'abc', outputPath: 'abc'},
+          {path: 'cde'},
+          {path: 'fgh', outputPath: 'fgh'},
         ];
         processor.$process(docs);
         expect(docs.pop().urls).toEqual(['abc', 'fgh']);
       });
 
-      it('ignoring blacklisted doc types', () => {
+      it('ignoring blocked doc types', () => {
```
[the remark this answers is not in the document]
Comment (IgorMinar) [reply aio/tools/transforms/angular-base-package/processors/createSitemap.spec.js]: +1

File: aio/tools/transforms/angular-base-package/processors/createSitemap.spec.js [hunk 9]
```diff
@@ -39,45 +35,45 @@ describe('createSitemap processor', () => {
...
+          {path: 'fgh', outputPath: 'fgh', docType: 'good'},
         ];
-        processor.blacklistedDocTypes = ['bad'];
+        processor.ignoredDocTypes = ['bad'];
         processor.$process(docs);
         expect(docs.pop().urls).toEqual(['abc', 'fgh']);
       });
 
-      it('ignoring blacklisted paths', () => {
+      it('ignoring blocked paths', () => {
```
[the remark this answers is not in the document]
Comment (IgorMinar) [reply aio/tools/transforms/angular-base-package/processors/createSitemap.spec.js]: +1

File: integration/cli-hello-world/src/polyfills.ts [hunk 10]
```diff
@@ -63,21 +63,26 @@
  *
  * The following flags will work for all browsers.
  *
- * (window as any).__Zone_disable_requestAnimationFrame = true; // disable patch requestAnimationFrame
+ * ```js
```
Comment (IgorMinar) [inline integration/cli-hello-world/src/polyfills.ts]: again. all of these changes should be upstreamed to CLI first and then downstreamed here. is that what you are doing?

File: packages/compiler/src/assertions.ts [hunk 11]
```diff
@@ -20,7 +20,7 @@ export function assertArrayOfStrings(identifier: string, value: any) {
   }
 }
 
-const INTERPOLATION_BLACKLIST_REGEXPS = [
+const FORBIDDEN_INTERPOLATION_REGEXPS = [
```
Comment (IgorMinar) [inline packages/compiler/src/assertions.ts]: how about: 
```
INTERPOLATION_BLOCKLIST = [...] : RegExp[]
```
?

File: packages/core/src/sanitization/url_sanitizer.ts [hunk 12]
```diff
@@ -36,9 +36,6 @@ import {isDevMode} from '../util/is_dev_mode';
  */
 const SAFE_URL_PATTERN = /^(?:(?:https?|mailto|ftp|tel|file):|[^&:/?#]*(?:[/?#]|$))/gi;
 
-/* A pattern that matches safe srcset values */
-const SAFE_SRCSET_PATTERN = /^(?:(?:https?|file):|[^&:/?#]*(?:[/?#]|$))/gi;
```
[the remark this answers is not in the document]
Comment (IgorMinar) [reply packages/core/src/sanitization/url_sanitizer.ts]: +1 - what happened?

it's concerning that this is in url sanitizer - can you please verify how this line was made unused and if the previous change was correct and only forgot to remove this line. We need to be confident that we didn't screw up the previous change before removing this line.

File: packages/platform-browser/src/dom/events/dom_events.ts [hunk 13]
```diff
@@ -38,19 +38,19 @@ const NATIVE_REMOVE_LISTENER = 'removeEventListener';
...
 const stopMethodSymbol = '__zone_symbol__stopImmediatePropagation';
 
-const blackListedEvents: string[] =
-    (typeof Zone !== 'undefined') && (Zone as any)[__symbol__('BLACK_LISTED_EVENTS')];
-let blackListedMap: {[eventName: string]: string};
-if (blackListedEvents) {
-  blackListedMap = {};
-  blackListedEvents.forEach(eventName => { blackListedMap[eventName] = eventName; });
+const disabledEvents: string[] =
+    (typeof Zone !== 'undefined') && (Zone as any)[__symbol__('DISABLE_EVENTS')];
```
[the remark this answers is not in the document]
Comment (IgorMinar) [reply packages/platform-browser/src/dom/events/dom_events.ts]: correct - you can't change this without requiring a new version of zone.js in peer deps. alternatively, and likely more preferably we should check for both symbols and leave a comment about when to remove the obsolete one in the future.

[the remark this answers is not in the document]
Comment (IgorMinar) [reply packages/platform-browser/src/dom/events/dom_events.ts]: also, shouldn't it be `DISABLED_EVENTS` ? or something even more descriptive?

File: packages/platform-browser/test/dom/events/event_manager_spec.ts (hunk 1 above)
Comment (IgorMinar) [reply packages/platform-browser/test/dom/events/event_manager_spec.ts]: +1

File: packages/platform-browser/testing/src/browser_util.ts (hunk 2 above)
Comment (IgorMinar) [reply packages/platform-browser/testing/src/browser_util.ts]: +1 for `_voidTags`

File: test-events.js (hunk 3 above)
Comment (IgorMinar) [reply test-events.js]: +1

File: tools/material-ci/angular_material_test_blocklist.js (hunk 4 above)
Comment (IgorMinar) [reply tools/material-ci/angular_material_test_blocklist.js]: whatever we do, we should be consistent about the capitalization across the board...

File: tools/ts-api-guardian/test/unit_test.ts [hunk 14]
```diff
@@ -357,28 +357,30 @@ describe('unit test', () => {
     check({'file.d.ts': input}, expected, {stripExportPattern: /^__.*/});
   });
 
-  it('should throw on using non-whitelisted module imports in expression position', () => {
+  it('should throw on using disallowed module imports in expression position', () => {
```
Comment (IgorMinar) [inline tools/ts-api-guardian/test/unit_test.ts]:
```suggestion
  it('should throw on using module import names in expression position, if the names have not been explicitly allowed', () => {
```

**Review by IgorMinar — COMMENTED** [review]
this PR needs some work but overall looks good. please clean it up.

File: aio/content/guide/browser-support.md [hunk 15]
```diff
@@ -604,7 +604,7 @@ If you are not using the CLI, you should add your polyfill scripts directly to t
      */
     // __Zone_disable_requestAnimationFrame = true; // disable patch requestAnimationFrame
     // __Zone_disable_on_property = true; // disable patch onProperty such as onclick
-    // __zone_symbol__BLACK_LISTED_EVENTS = ['scroll', 'mousemove']; // disable patch specified eventNames
+    // __zone_symbol_DISABLE_EVENTS = ['scroll', 'mousemove']; // disable patch specified eventNames
```
[the remark this answers is not in the document]
Comment (benlesh) [reply aio/content/guide/browser-support.md]: cc @JiaLiPassion it seems like these things must be upstreamed in Zone.js?

File: aio/tools/examples/shared/boilerplate/cli/src/polyfills.ts (hunk 5 above)
Comment (benlesh) [reply aio/tools/examples/shared/boilerplate/cli/src/polyfills.ts]: The comments here seems largely duplicated through more than one copy of `polyfills.ts`.  I may have copied and pasted improperly here.

File: aio/tools/transforms/angular-base-package/processors/createSitemap.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (benlesh) [reply aio/tools/transforms/angular-base-package/processors/createSitemap.js]: Igor is correct, this was auto-formatted.

File: aio/tools/transforms/angular-base-package/processors/createSitemap.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (benlesh) [reply aio/tools/transforms/angular-base-package/processors/createSitemap.js]: Auto-formatted by clang formatter (our formatter)

File: aio/content/guide/browser-support.md (hunk 15 above)
[the remark this answers is not in the document]
Comment (gkalpak) [reply aio/content/guide/browser-support.md]: If we want to make these changes in zone.js, they either need to happen in a backwards compatible way or via a breaking change (which means Angular also needs a breaking change to break compatibility with older zone.js versions).

[the remark this answers is not in the document]
Comment (JiaLiPassion) [reply aio/content/guide/browser-support.md]: @benlesh, @gkalpak, yeah, I can make a PR in `zone.js` to do the rename and also keep the `backward compatiblity` so the `__zone_symbol__BLACK_LISTED_EVENTS` and `__zone_symbol_DISABLE_EVENTS` can all work.

File: packages/platform-browser/src/dom/events/dom_events.ts (hunk 13 above)
[the remark this answers is not in the document]
Comment (JiaLiPassion) [reply packages/platform-browser/src/dom/events/dom_events.ts]: Like above, I will make a PR in `zone.js` first.

File: test-events.js (hunk 3 above)
Comment (JiaLiPassion) [reply test-events.js]: @skreborn , @IgorMinar , sure, got it!

**Comment by benlesh** [comment]
Most of these issues were resolved in other PRs. I broke the remaining issues into smaller PRs. We can close this one.

**Comment by elimin8r** [comment]
seems like a lot of effort to rename some variables. also, if you dont scrub the git history, the very beings you are catering to will still be triggered when they examine the past, thus rendering this an exercise in futility.

**Pull request closed without merging**